### DISCUSSION TITLE
Does interactiveLayerIds need to be defined now to access event.features

### DISCUSSION THREAD

**Discussion by evrazi** [uber/react-map-gl#2074] [Q&A]
Just migrated from v6 where this return all layers that a user clicked on, this same code produces undefined now.

```
onClick={(e) => {
  console.log(e.features);
}}
```

Tried this too, with no luck
`
interactiveLayerIds = {['*']}
`

**Comment by RobJJ** [comment]
Did you manage to get interactiveLayerIds to work as intended? When I try access the features, the array is always empty... not sure why its not referencing the layer "id" features provided.

**Comment by evrazi** [comment]
@RobJJ Looking at the code now it looks like you need to specify which ids you want to make interactive by passing ids to the<ReactMapGL> comp through _interactiveLayerIds_.

```
      <ReactMapGL interactiveLayerIds={arrayOfIds}>
```
[https://visgl.github.io/react-map-gl/docs/api-reference/map#interactivelayerids](https://visgl.github.io/react-map-gl/docs/api-reference/map#interactivelayerids)

Please take this with a grain of salt. I haven't worked on this project in almost a year and some things could have changed by now.

Hope this helps :)

**Reply by RobJJ** [reply]
Thanks @evrazi , I got it working. I had miss understood this sentence in the docs "listeners will be triggered only if its location is within a visible feature"
